tcms/testruns/tests/test_permissions.py

```diff
@@ -0,0 +1,63 @@
+from django.urls import reverse
+from http import HTTPStatus
+
+from tcms import tests
+from tcms.tests import factories
+from tcms.testcases.models import TestCaseStatus
+
+
+class TestCreateTestRun(tests.PermissionsTestCase):
+    permission_label = 'testruns.add_testrun'
+    url = reverse('testruns-new')
+    http_method_names = ['get', 'post']
+
+    @classmethod
+    def setUpTestData(cls):
+
+        plan = factories.TestPlanFactory()
+        build = factories.BuildFactory()
+
+        cls.post_data = {
+            'summary': plan.pk,
+            'from_plan': plan.pk,
+            'build': build.pk,
+            'notes': 'Clone new run',
+            'POSTING_TO_CREATE': 'YES',
+        }
+
+        super().setUpTestData()
+
+        # cls.tester is created after calling super() above
+        cls.post_data['manager'] = cls.tester.email
+        cls.post_data['default_tester'] = cls.tester.email
+
+        cls.case_status_confirmed = TestCaseStatus.objects.get(name='CONFIRMED')
+
+        case_1 = factories.TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=cls.case_status_confirmed,
+            plan=[plan])
+        case_1.save()  # will generate history object
+
+        case_2 = factories.TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=cls.case_status_confirmed,
+            plan=[plan])
+        case_2.save()  # will generate history object
+
+        cls.post_data['case'] = (case_1.pk, case_2.pk)
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+
+    def verify_post_with_permission(self):
+
+        response = self.client.post(self.url, self.post_data, follow=True)
+
+        self.assertEqual(HTTPStatus.OK, response.status_code)
```

Comment:
See what `bugs/tests/test_permissions.py::NewTestCase.verify_post_with_permission` does as well as 
`testruns/tests/test_views.py::TestCreateNewRun.test_create_a_new_run()`. You have partially copied some of that code and left the rest.

You will have to copy all of the asserts and adjust the test case accordingly.

Notice how the original test case doesn't use `follow=True`

Reply:
Note: this view does support 2 modes of operation based on POSTING_TO_CREATE variable. I need to examine this in more details to see what is going on there.